azure/azure.go

```diff
@@ -1152,3 +1164,19 @@ func isExponentialError(err error) bool {
 	}
 	return false
 }
+
+func azureBaseURI(cloudEnvName string) (string, error) {
+	if value, ok := os.LookupEnv(auth.EnvironmentName); ok {
```

Comment:
Ideally we don't need this as we get the azure cloud environment from the instance metadata. Keeping this in case the instance metadata (older versions did not have the azEnvironment metadata variable) does not have the field and user wants to explicitly override it.